Library/OcMainLib/OpenCoreKernel.c

```diff
@@ -1171,6 +1206,31 @@ OcKernelFileOpen (
     && StrCmp (FileName, L"System\\Library\\Kernels\\kernel") != 0
     && OcStriStr (FileName, L".kext\\") == NULL
     && OcStriStr (FileName, L".im4m") == NULL) {
+    //
+    // Change the target to the custom one if requested CustomKernel.
+    //
+    if (mCustomKernelDirectory != NULL) {
+      DEBUG ((DEBUG_INFO, "OC: Redirecting %s to the custom one on ESP\n", FileName));
+      NewFileName = OcStrrChr (FileName, L'\\');
+      if (NewFileName == NULL) {
+        NewFileName = FileName;
+      }
+
+      DEBUG ((DEBUG_INFO, "OC: Filename after redirection: %s\n", NewFileName));
+
+      mCustomKernelDirectoryInProgress = TRUE;
+      Status = SafeFileOpen (mCustomKernelDirectory, &EspNewHandle, NewFileName, OpenMode, Attributes);
+      mCustomKernelDirectoryInProgress = FALSE;
+      if (!EFI_ERROR (Status)) {
+        if (*NewHandle != NULL) {
```

Comment:
This `if` is unnecessary, as there was an `EFI_ERROR (Status)` check above with a `return`. `*NewHandle` is guaranteed to be set. Notably checking it for `NULL` here is not correct in the first place, as the handle may not be initialised on failure.